db/blip_sync_context.go

```diff
@@ -34,7 +34,10 @@ const (
 
 var ErrClosedBLIPSender = errors.New("use of closed BLIP sender")
 
-func NewBlipSyncContext(ctx context.Context, bc *blip.Context, db *Database, contextID string, replicationStats *BlipSyncStats) *BlipSyncContext {
+func NewBlipSyncContext(ctx context.Context, bc *blip.Context, db *Database, contextID string, replicationStats *BlipSyncStats, ctxCancelFunc context.CancelFunc) (*BlipSyncContext, error) {
+	if ctxCancelFunc == nil {
```

Comment:
Is this required for cases where the BlipSyncContext is for an active peer (ISGR)?  Closing the connection in that case feels like a different problem than the passive peer case.